tensorflow/core/ops/mkl_nn_ops.cc

```diff
@@ -804,6 +803,208 @@ REGISTER_OP("_MklDepthwiseConv2dNativeBackpropFilter")
+    .Attr(GetPaddingAttrString())
+    .Attr("dilations: list(int) = [1, 1, 1, 1]")
+    .SetShapeFn([](InferenceContext* c) {
+      TF_RETURN_IF_ERROR(shape_inference::Conv2DShape(c));
+      ShapeHandle unused, channel;
+      TF_RETURN_IF_ERROR(c->WithRank(c->input(2), 1, &unused));
+      TF_RETURN_IF_ERROR(c->WithRank(c->input(3), 0, &unused));
+      TF_RETURN_IF_ERROR(c->WithRank(c->input(4), 0, &unused));
+      TF_RETURN_IF_ERROR(c->WithRankAtMost(c->input(5), 1, &channel));
+      TF_RETURN_IF_ERROR(c->WithRankAtMost(c->input(6), 1, &channel));
+      c->set_output(1, channel);
+      c->set_output(2, channel);
+      return Status::OK();
+    })
+    .Doc(R"doc(
+MKL-DNN implementation of quantized depthwise Conv2D with Bias.
+*NOTE*: Do not invoke this operator directly in Python. Graph rewrite pass is
+expected to invoke this operator.
+)doc");
+
+REGISTER_OP("_MklQuantizedDepthwiseConv2DWithBiasAndRelu")
+    .Input("input: Tinput")
+    .Input("filter: Tfilter")
+    .Input("bias: float")
+    .Input("min_input: float")
+    .Input("max_input: float")
+    .Input("min_filter: float")
+    .Input("max_filter: float")
+    .Input("mkl_input: uint8")
+    .Input("mkl_filter: uint8")
+    .Input("mkl_bias: uint8")
+    .Input("mkl_min_input: uint8")
+    .Input("mkl_max_input: uint8")
+    .Input("mkl_min_filter: uint8")
+    .Input("mkl_max_filter: uint8")
+    .Output("output: out_type")
+    .Output("min_output: float")
+    .Output("max_output: float")
+    .Output("mkl_output: uint8")
+    .Output("mkl_min_output: uint8")
+    .Output("mkl_max_output: uint8")
+    .Attr("Tinput: quantizedtype")
+    .Attr("Tfilter: quantizedtype")
+    // Additional attribute "T" for enabling MKL to TF conversion
+    .Attr("T: quantizedtype")
+    .Attr("out_type: quantizedtype = DT_QINT32")
+    .Attr("data_format: string = 'NHWC'")
+    .Attr("strides: list(int)")
+    .Attr("is_filter_const: bool = true")
+    .Attr(GetPaddingAttrString())
+    .Attr("dilations: list(int) = [1, 1, 1, 1]")
+    .SetShapeFn([](InferenceContext* c) {
+      TF_RETURN_IF_ERROR(shape_inference::Conv2DShape(c));
+      ShapeHandle unused, channel;
+      TF_RETURN_IF_ERROR(c->WithRank(c->input(2), 1, &unused));
+      TF_RETURN_IF_ERROR(c->WithRank(c->input(3), 0, &unused));
+      TF_RETURN_IF_ERROR(c->WithRank(c->input(4), 0, &unused));
+      TF_RETURN_IF_ERROR(c->WithRankAtMost(c->input(5), 1, &channel));
+      TF_RETURN_IF_ERROR(c->WithRankAtMost(c->input(6), 1, &channel));
+      c->set_output(1, channel);
+      c->set_output(2, channel);
+      return Status::OK();
+    })
+    .Doc(R"doc(
+MKL-DNN implementation of quantized depthwise Conv2D with Bias and Relu.
+*NOTE*: Do not invoke this operator directly in Python. Graph rewrite pass is
+expected to invoke this operator.
+)doc");
+
+REGISTER_OP("_MklQuantizedDepthwiseConv2DWithBiasAndReluAndRequantize")
+    .Input("input: Tinput")
+    .Input("filter: Tfilter")
+    .Input("bias: float")
```

Comment:
@bhavani-subramanian need to change `.Input("bias: float")` -> .Input("bias: Tbias")`

Reply:
The API review is happening this afternoon so we are pulling to create an internal changelist now. This change is not related to API and can be added to the changelist later (through this PR). Or I can make the change directly as a separate commit as well.

Reply:
@penpornk Could you please make this change as a separate commit? Thanks.

Reply:
Sure, I'll do that after this PR is merged. (If it gets through the API review.)

Reply:
Thanks @penpornk!

Reply:
Change submitted in https://github.com/tensorflow/tensorflow/commit/978dba096289fd03b9baa38c82e966eba03c6e1a.